pkg/flowaggregator/kafka/convertor.go

```diff
@@ -0,0 +1,142 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"net"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/kafka/producer/convertor"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/flowaggregator/kafka/protobuf"
+)
+
+const (
+	AntreaFlowMsg string = "AntreaFlowMsg"
+	AntreaTopic   string = "AntreaTopic"
+)
+
+// convertRecordToAntreaFlowMsg is to support the AntreaFlowMsg proto schema.
+type convertRecordToAntreaFlowMsg struct{}
+
+func NewAntreaFlowMsgConvertor() convertor.IPFIXToKafkaConvertor {
+	return &convertRecordToAntreaFlowMsg{}
+}
+
+func (c *convertRecordToAntreaFlowMsg) ConvertIPFIXMsgToFlowMsgs(msg *entities.Message) []protoreflect.Message {
+	convertRecordToFlowMsg := func(msg *entities.Message, record entities.Record) protoreflect.Message {
+		antreaFlowMsg := &protobuf.AntreaFlowMsg{}
+		antreaFlowMsg.TimeReceived = msg.GetExportTime()
+		antreaFlowMsg.SequenceNumber = msg.GetSequenceNum()
+		antreaFlowMsg.ObsDomainID = msg.GetObsDomainID()
+		antreaFlowMsg.ExportAddress = msg.GetExportAddress()
+		addAllFieldsToAntreaFlowMsg(antreaFlowMsg, record)
+		return antreaFlowMsg.ProtoReflect()
+	}
+	// Convert all records in IPFIX set to Flow messages.
+	set := msg.GetSet()
+	if set.GetSetType() == entities.Template {
+		return nil
+	}
+	records := set.GetRecords()
+	flowMsgs := make([]protoreflect.Message, len(records))
+	for i, record := range records {
+		flowMsgs[i] = convertRecordToFlowMsg(msg, record)
+	}
+	return flowMsgs
+}
+
+func (c *convertRecordToAntreaFlowMsg) ConvertIPFIXRecordToFlowMsg(record entities.Record) protoreflect.Message {
+	// We add the fields from the record directly and do not add the fields from
+	// the IPIFX message to antreaFlowMsg kafka flow message.
+	antreaFlowMsg := &protobuf.AntreaFlowMsg{}
+	addAllFieldsToAntreaFlowMsg(antreaFlowMsg, record)
+
+	return antreaFlowMsg.ProtoReflect()
+}
+
+func addAllFieldsToAntreaFlowMsg(flowMsg *protobuf.AntreaFlowMsg, record entities.Record) {
+	for _, ie := range record.GetOrderedElementList() {
+		switch ie.Element.Name {
+		case "flowStartSeconds":
+			flowMsg.TimeFlowStartInSecs = ie.Value.(uint32)
+		case "flowEndSeconds":
+			flowMsg.TimeFlowEndInSecs = ie.Value.(uint32)
+		case "sourceIPv4Address", "sourceIPv6Address":
+			if flowMsg.SrcIP != "" {
+				klog.Warningf("Do not expect source IP: %v to be filled already", flowMsg.SrcIP)
+			}
+			flowMsg.SrcIP = ie.Value.(net.IP).String()
+		case "destinationIPv4Address", "destinationIPv6Address":
+			if flowMsg.DstIP != "" {
+				klog.Warningf("Do not expect destination IP: %v to be filled already", flowMsg.DstIP)
```

Comment:
as part of the transition to structured logging, I don't think we use `klog.Warningf`, but just `klog.InfoS`. @tnqn is that correct?